### PR TITLE
Improve error masking

### PR DESCRIPTION
**Pull request by rossf7**
We had some problems yesterday tracking down a too many results error. I found the same problem in a few other places.

Adding more details to make debugging easier.

### CODE REVIEW

File: service/awsconfig/v4/resource/cloudformation/adapter/host_route_tables.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func waitForPeeringConnectionID(cfg Config) (string, error) {
 			return microerror.Mask(err)
 		}
 		if len(output.VpcPeeringConnections) > 1 {
-			return microerror.Mask(tooManyResultsError)
+			return microerror.Maskf(tooManyResultsError, "too many peering connections")
```
Comment (kopiczko) [inline service/awsconfig/v4/resource/cloudformation/adapter/host_route_tables.go]: The error message will be like: `too many results: too many peering connections`. Looks like too many too manies :D

File: service/awsconfig/v4/resource/cloudformation/adapter/host_route_tables.go [hunk 2]
```diff
@@ -120,7 +120,7 @@ func routeTableID(name string, cfg Config) (string, error) {
 		return "", microerror.Mask(err)
 	}
 	if len(output.RouteTables) > 1 {
-		return "", microerror.Mask(tooManyResultsError)
+		return "", microerror.Maskf(tooManyResultsError, "too many route tables: %s", name)
```
Comment (kopiczko) [inline service/awsconfig/v4/resource/cloudformation/adapter/host_route_tables.go]: Similar to above.

File: service/awsconfig/v4/resource/endpoints/instance.go [hunk 3]
```diff
@@ -43,10 +43,10 @@ func (r Resource) findMasterInstance(instanceName string) (*ec2.Instance, error)
 	}
 
 	if instancesFound < 1 {
-		return nil, microerror.Mask(notFoundError)
+		return nil, microerror.Maskf(notFoundError, "could not find instance: %s", instanceName)
```
Comment (kopiczko) [inline service/awsconfig/v4/resource/endpoints/instance.go]: Here `not found: could not find instance ...`

File: service/awsconfig/v4/resource/endpoints/instance.go [hunk 4]
```diff
@@ -43,10 +43,10 @@ func (r Resource) findMasterInstance(instanceName string) (*ec2.Instance, error)
...
 	}
 	if instancesFound > 1 {
-		return nil, microerror.Mask(tooManyResultsError)
+		return nil, microerror.Maskf(tooManyResultsError, "too many instances: %s", instanceName)
```
Comment (kopiczko) [inline service/awsconfig/v4/resource/endpoints/instance.go]: too many

File: service/awsconfig/v4/resource/cloudformation/validate.go [hunk 5]
```diff
@@ -44,7 +44,7 @@ func (r *Resource) validateHostPeeringRoutes(cluster v1alpha1.AWSConfig) error {
 	}
 	output, err := r.HostClients.EC2.DescribeRouteTables(input)
 	if err == nil && len(output.RouteTables) == 1 {
-		return microerror.Mask(existingRouteError)
+		return microerror.Maskf(existingRouteError, "route already exists: %s", key.PrivateSubnetCIDR(cluster))
```
Comment (kopiczko) [inline service/awsconfig/v4/resource/cloudformation/validate.go]: Here error var is a bit too detailed. I try to encourage to create more generic errors.

Right now it's `existing route: route already exists 'name'`.

I'd change that to `alreadyExistsError`, and then `already exists: route 'name'`.
So generic error + details.

Comment (rossf7) [reply service/awsconfig/v4/resource/cloudformation/validate.go]: Good point, done.

**Comment by kopiczko** [comment]
Yeah, looks awesome now <3

**Pull request merged**